connect-inject/health_check_resource.go

```diff
@@ -166,7 +171,10 @@ func (h *HealthCheckResource) reconcilePod(pod *corev1.Pod) error {
 		// Create a new health check.
 		h.Log.Debug("registering new health check", "name", pod.Name, "id", healthCheckID)
 		err = h.registerConsulHealthCheck(client, healthCheckID, serviceID, status)
-		if err != nil {
+		if errors.Is(err, ServiceNotFoundErr) {
+			h.Log.Warn("skipping registration because service not registered with Consul-this may be because the pod is shutting down", "serviceID", serviceID)
```

Comment:
```suggestion
			h.Log.Warn("skipping registration because service not registered with Consul - this may be because the pod is shutting down", "serviceID", serviceID)
```